File: 0108-convert-sorted-array-to-binary-search-tree/0108-convert-sorted-array-to-binary-search-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public TreeNode fun(int[] nums,int low,int high){
        // Base Case lelo
        if(low>high){
            return null;
        }
        // Root middle wla element hi bnta hai in a Sorted Array
        int mid=(low+high)/2;

        TreeNode root=new TreeNode(nums[mid]);

        // Left Child bna lo
        root.left=fun(nums,low,mid-1);          // Mid element k Left wley Tree mei Left mei aaengey

        // Right Child bna lo
        root.right=fun(nums,mid+1,high);        // Mid element k next element se lekey High tak Right Tree aaega

        return root;
    }

    public TreeNode sortedArrayToBST(int[] nums) {
        // Recursive Function Call karlo
        return fun(nums,0,nums.length-1);
    }
}

File: 0108-convert-sorted-array-to-binary-search-tree/0108-convert-sorted-array-to-binary-search-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
TreeNode*fun(vector<int>&nums,int low,int high){
    // Base Case lelo 
    if(low>high){
        return nullptr;
    }
    // Root middle wla element hi bnta hai in a sorted array 
    int mid=(low+high)/2;

    TreeNode*root=new TreeNode(nums[mid]); 

    // Left Child bnaq hai 
    root->left=fun(nums,low,mid-1);         // Mid element k left wley tree mei left  mei aaengey 

    // Right Child bna loo
    root->right=fun(nums,mid+1,high);       // Mid element k ext element se lekey high tak right tree aaega 

    return root;
}

    TreeNode* sortedArrayToBST(vector<int>& nums) {
        // Recursive Function call karloo
        return fun(nums,0,nums.size()-1);
    }
};